### ISSUE TITLE
Tests for 2.0-rc1

### ISSUE THREAD

**Issue by sorinsarca** [rogervila/array-diff-multidimensional#12]
Here are some tests

Numbers

```php
$a = [1];
$b = [PHP_FLOAT_EPSILON + 1];


$r = ArrayDiffMultidimensional::compare($a, $b);
// expected: [1], actual: []
var_dump($r);
```

Arrays

```php
$a = [[]];
$b = [1];


$r = ArrayDiffMultidimensional::compare($a, $b);
// expected: [[]], actual: []
var_dump($r);
```

References

```php
$a = [1, 2, &$a];
$b = [1, 2, &$b];

$r = ArrayDiffMultidimensional::compare($a, $b);
// expected: [], actual: infinite recursion
var_dump($r);
```

**Comment by rogervila** [comment]
Thank you @sorinsarca! 
I will add them on the test suite

**Comment by rogervila** [comment]
The references issue is quite a corner case, and since I do not have a lot of time I think I will not add support for it.

I suggest to use `ArrayObject` to create a copy of the array before passing it to the `compare` call: https://www.php.net/manual/es/arrayobject.getarraycopy.php

**Comment by rogervila** [comment]
I have released 2.0.0
Thank you for opening this issue.
